pkg/ipmi/oem.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// OEM implementation
+package ipmi
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+var (
+	productList = [2]string{"Tioga Pass", "Mono Lake"}
+)
+
+func getSystemProductName(si *smbios.Info) (string, error) {
+	t1, err := si.GetSystemInformation()
+	if err != nil {
+		return "", err
+	}
+	return t1.ProductName, nil
+}
+
+func isMatched(productName string) bool {
+	for _, v := range productList {
+		if strings.HasPrefix(productName, v) {
+			return true
+		}
+	}
+	return false
+}
+
+func (i *IPMI) RunOEMCommands(si *smbios.Info) error {
+	productName, err := getSystemProductName(si)
+	if err != nil {
+		return err
+	}
+	if isMatched(productName) {
```

Comment:
Is there any reason for this conditional to be here? I think the uinit is better fitted for this and in the future we should probably have a customizable uinit.

Also, I think the IPMI package should only have an API for sending and receiving info, not for gathering memory or cpu info if that was the plan.

Reply:
I was thinking putting product names to pkg/ipmp/oem.go and keep uinit generic, I will move the isMatched() and the productList array to uinit. 
The OEM IPMI commands will send machine information (DIMM, CPU info., etc) the BMC, should those gathering info code be done in uinit/oem.go or in pkg/ipmi/oem.go?

Reply:
I am not sure what the best place is but I would decouple the data gathering from the IPMI send commands.

Reply:
Updated.